integration/test/Test/Conversation.hs

```diff
@@ -177,3 +179,136 @@ testCreateConversationNonFullyConnected = do
+          >=> removeField "optSettings.setFederationDomainConfigs"
+          >=> setField "optSettings.setFederationDomainConfigsUpdateFreq" (Aeson.Number 1)
+  startDynamicBackends
+    [ def {dbBrig = setFederationConfig},
+      def {dbBrig = setFederationConfig}
+    ]
+    $ \dynDomains -> do
+      domains@[domainA, domainB] <- pure dynDomains
+      connectAllDomainsAndWaitToSync 1 domains
+      [uA, uB] <- createAndConnectUsers [domainA, domainB]
+      withWebSocket uA $ \ws -> do
+        -- create group conversation owned by domainB
+        convId <- bindResponse (postConversation uB (defProteus {qualifiedUsers = [uA]})) $ \r -> do
+          r.status `shouldMatchInt` 201
+          r.json %. "qualified_id"
+
+        -- check conversation exists and uB is a member from POV of uA
+        bindResponse (getConversation uA convId) $ \r -> do
+          r.status `shouldMatchInt` 200
+          members <- r.json %. "members.others" & asList
+          qIds <- for members (\m -> m %. "qualified_id")
+          uBQId <- objQidObject uB
+          qIds `shouldMatchSet` [uBQId]
+
+        -- check conversation exists and uA is a member from POV of uB
+        bindResponse (getConversation uB convId) $ \r -> do
+          r.status `shouldMatchInt` 200
+          members <- r.json %. "members.others" & asList
+          qIds <- for members (\m -> m %. "qualified_id")
+          uAQId <- objQidObject uA
+          qIds `shouldMatchSet` [uAQId]
+
+        -- domainA stops federating with domainB
+        void $ Internal.deleteFedConn domainA domainB
+
+        -- assert conversation deleted from domainA
+        retryT $
+          bindResponse (getConversation uA convId) $ \r ->
+            r.status `shouldMatchInt` 404
+
+        -- assert federation.delete event is sent twice
+        void $ awaitNMatches 2 3 (\n -> nPayload n %. "type" `isEqual` "federation.delete") ws
+
+      -- assert no conversation.delete event is sent to uA
+      eventPayloads <-
+        getNotifications uA "cA" def
+          >>= getJSON 200
+          >>= \n -> n %. "notifications" & asList >>= \ns -> for ns nPayload
+
+      forM_ eventPayloads $ \p ->
+        p %. "type" `shouldNotMatch` "conversation.delete"
+
+testDefederationOneOnOne :: HasCallStack => App ()
+testDefederationOneOnOne = do
+  let setFederationConfig =
+        setField "optSettings.setFederationStrategy" "allowDynamic"
+          >=> removeField "optSettings.setFederationDomainConfigs"
+          >=> setField "optSettings.setFederationDomainConfigsUpdateFreq" (Aeson.Number 1)
+  startDynamicBackends
+    [ def {dbBrig = setFederationConfig},
+      def {dbBrig = setFederationConfig}
+    ]
+    $ \dynDomains -> do
+      domains@[domainA, domainB] <- pure dynDomains
+      connectAllDomainsAndWaitToSync 1 domains
+      [uA, uB] <- createAndConnectUsers [domainA, domainB]
+      -- figure out on which backend the 1:1 conversation is created
+      conn <- getConnection uA uB
```

Comment:
`conn` usually refers to a cannon web socket connection of one client, but i don't have a better name either.  maybe this is fine?  (`uconn`?)